components/camel-kamelet/src/main/java/org/apache/camel/component/kamelet/Kamelet.java

```diff
@@ -116,6 +116,10 @@ public static RouteDefinition templateToRoute(RouteTemplateDefinition in, Map<St
         RouteDefinition def = in.asRouteDefinition();
         def.setId(rid);
 
+        if (def.getInput() == null) {
+            throw new IllegalArgumentException("Camel route " + rid + " input does not exist.");
```

Comment:
How's this wording for the error message?

Reply:
Yes this is fine